validator/sawtooth_validator/consensus/proxy.py

```diff
@@ -29,6 +33,65 @@ class UnknownBlock(Exception):
     """The given block could not be found."""
 
 
+class ConsensusActivationObserver(ChainObserver):
+    """Observes chain updates and activates new engines, if they have been activated."""
+
+    def __init__(self, consensus_registry, consensus_notifier, settings_view_factory):
+        self._registry = consensus_registry
+        self._notifier = consensus_notifier
+        self._settings_view_factory = settings_view_factory
+
+    def chain_update(self, block, receipts):
+        # Check to see if there is an active engine at all:
+        if not self._registry.has_active_engine():
+            self._update_active_engine(block)
+            return
+
+        # Otherwise, check to see fi the settings changed, before updating
+        # the active engine.
+        receipt_events = ReceiptEventExtractor(receipts).extract([
+            EventSubscription(event_type="setting/update")])
+
+        consensus_settings_changed = False
+        for event in receipt_events:
+            consensus_settings_changed = self._is_consensus_algorithm_setting(event)
+            if consensus_settings_changed:
+                break
+
+        if consensus_settings_changed:
+            self._update_active_engine(block)
+
+    def _update_active_engine(self, block):
+        header = BlockHeader()
+        header.ParseFromString(block.header)
+        settings_view = self._settings_view_factory.create_settings_view(
+            header.state_root_hash)
+
+        conf_name = settings_view.get_setting(
+            'sawtooth.consensus.algorithm.name')
+        conf_version = settings_view.get_setting(
+            'sawtooth.consensus.algorithm.version')
+
+        if not self._registry.is_active_engine_name_version(conf_name, conf_version):
+            self._registry.activate_engine(conf_name, conf_version)
+
+            self._notifier.notify_engine_activated(block)
```

Comment:
If already is an active engine, it should be notified that is deactivated (see how the chain controller currently does this)

Reply:
fixed